src/Shared/BrowserTesting/src/BrowserManagerConfiguration.cs

```diff
@@ -0,0 +1,366 @@
+                StorageStatePath = overrideOptions?.StorageStatePath != default ? overrideOptions.StorageStatePath : defaultOptions.StorageStatePath,
+                StorageState = overrideOptions?.StorageState != default ? overrideOptions.StorageState : defaultOptions.StorageState
+            };
+
+        private LaunchOptions Combine(LaunchOptions defaultOptions, LaunchOptions overrideOptions) =>
+            new()
+            {
+                IgnoreDefaultArgs = overrideOptions.IgnoreDefaultArgs != default ? overrideOptions.IgnoreDefaultArgs : defaultOptions.IgnoreDefaultArgs,
+                ChromiumSandbox = overrideOptions.ChromiumSandbox != default ? overrideOptions.ChromiumSandbox : defaultOptions.ChromiumSandbox,
+                HandleSIGHUP = overrideOptions.HandleSIGHUP != default ? overrideOptions.HandleSIGHUP : defaultOptions.HandleSIGHUP,
+                HandleSIGTERM = overrideOptions.HandleSIGTERM != default ? overrideOptions.HandleSIGTERM : defaultOptions.HandleSIGTERM,
+                HandleSIGINT = overrideOptions.HandleSIGINT != default ? overrideOptions.HandleSIGINT : defaultOptions.HandleSIGINT,
+                IgnoreAllDefaultArgs = overrideOptions.IgnoreAllDefaultArgs != default ? overrideOptions.IgnoreAllDefaultArgs : defaultOptions.IgnoreAllDefaultArgs,
+                SlowMo = overrideOptions.SlowMo != default ? overrideOptions.SlowMo : defaultOptions.SlowMo,
+                Env = overrideOptions.Env != default ? overrideOptions.Env : defaultOptions.Env,
+                DumpIO = overrideOptions.DumpIO != default ? overrideOptions.DumpIO : defaultOptions.DumpIO,
+                IgnoreHTTPSErrors = overrideOptions.IgnoreHTTPSErrors != default ? overrideOptions.IgnoreHTTPSErrors : defaultOptions.IgnoreHTTPSErrors,
+                DownloadsPath = overrideOptions.DownloadsPath != default ? overrideOptions.DownloadsPath : defaultOptions.DownloadsPath,
+                ExecutablePath = overrideOptions.ExecutablePath != default ? overrideOptions.ExecutablePath : defaultOptions.ExecutablePath,
+                Devtools = overrideOptions.Devtools != default ? overrideOptions.Devtools : defaultOptions.Devtools,
+                UserDataDir = overrideOptions.UserDataDir != default ? overrideOptions.UserDataDir : defaultOptions.UserDataDir,
+                Args = overrideOptions.Args != default ? overrideOptions.Args : defaultOptions.Args,
+                Headless = overrideOptions.Headless != default ? overrideOptions.Headless : defaultOptions.Headless,
+                Timeout = overrideOptions.Timeout != default ? overrideOptions.Timeout : defaultOptions.Timeout,
+                Proxy = overrideOptions.Proxy != default ? overrideOptions.Proxy : defaultOptions.Proxy
+            };
+    }
+
+    public record BrowserOptions(BrowserKind BrowserKind, LaunchOptions BrowserLaunchOptions, BrowserContextOptions DefaultContextOptions);
+}
```

Comment:
I don't really have a good sense of why so much config infrastructure is needed here. What kinds of scenarios do you have in mind for when the config would need to be tweaked at such a fine-grained level, and has to come from external config files?

> Also I haven't made this foolproof, I wrote it based on what I wanted us to be able to do through config and I stopped after I got things working.

That seems related. I'm curious about what things you want us to be able to do through config (which wouldn't work just to be hardcoded in C# sources).

Reply:
I don't want us to be hardcoding anything in C# sources because then we end up with N different configurations across the board, I rather take "the plunge" and write the mapping one time, which won't be hard to maintain tbh.

As to this, I expect us to be able to define a default config for all the browsers, specialize it per different browser instance (chrome, safari, firefox, or maybe we want to be able to have special settings for launching edge)

At the same time I want to make sure we can define browser contexts that we can reuse across the board, like if we want to test against a smaller viewport or things like that.